src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.PackTool.targets

```diff
@@ -0,0 +1,33 @@
+<!--
+***********************************************************************************************
+Microsoft.NET.PackTool.targets
+
+WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and have
+          created a backup copy.  Incorrect changes to this file will make it
+          impossible to load or build your projects from the command-line or the IDE.
+
+Copyright (c) .NET Foundation. All rights reserved. 
+***********************************************************************************************
+-->
+
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <Target Name="PackTool" DependsOnTargets="Publish" Condition=" '$(PackAsTool)' == 'true' ">
+    <ItemGroup>
+      <TfmSpecificPackageFiles Include="$(PublishDir)/**/*.*"/>
```

Comment:
I think we should be using the files that publish resolved and not globbing the disk. For one thing,  it's faster (no need to reprobe the disk). For another, it won't pack files that are laying around in your publish dir when you're developing in non-pristine state.

Reply:
I cannot find a complete itemgroup of all the files. Runtimeconfig is added the latest after resolving the files. Do you think I should use the list and append runtimeconfig to it? Sounds flaky tho, what if there is a more files like runtimeconfig

Reply:
maybe it could also skip the `CopyFilesToPublishDirectory` step by only depending on the targets required to compute the right files

Reply:
Looks like deps is also separate. I still think you should use the list + those two special cases because nothing else works by probing around in output paths like that. It will be weird that manually copying something in to the publish directory causes it to be packed in tool case, but manually copying to output directory does not cause it to be packed in normal case. 

If we miss a new special case, it's just a bug that's easy to fix, but globbing implies by-design flaky behavior when the publish directory is dirty.